sdk/dotnet/Pulumi.Kafka.csproj

```diff
@@ -44,7 +45,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Pulumi" Version="3.*" />
+    <PackageReference Include="Pulumi" Version="[3.54.1.0,4)" />
```

Comment:
Why is this now a fully specified version?

Reply:
@danielrbradley any idea if this is meaningful? that's a very old version to resolve to.

Reply:
Yup - this is what we should be including. The codegen knows that it references certain functions from the core SDK which weren't included before version 3..54.1. This minimum number also sometimes gets bumped as a new codegen version is released if a new feature is required from the core SDK which is why we want to avoid setting our own lax version ranges and instead defer to codegen where possible.

Reply:
> Why is this now a fully specified version?

This is a range saying `>= 3.54.1.0` and `< 4`.

Reply:
Ah that makes sense. Thanks!

Reply:
I've reverted this for now as it actually causes different resolution rules - using a range makes NuGet select the _minimum_ valid version rather than the latest 🤷